cobalt/media/player/web_media_player_impl.cc

```diff
@@ -972,6 +1075,9 @@ WebMediaPlayerClient* WebMediaPlayerImpl::GetClient() {
 void WebMediaPlayerImpl::OnDurationChanged() {
   if (ready_state_ == WebMediaPlayer::kReadyStateHaveNothing) return;
 
+  if (state_.is_prerolled && state_.is_demuxer_underflow) {
```

Comment:
`SourceBuffer.remove()` may lead to underflow, which is excluded here.

Reply:
Yes, it's intended, to keep the change as simply as possible.